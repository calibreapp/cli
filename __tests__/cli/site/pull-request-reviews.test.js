import {
  runCLI,
  setupIntegrationServer,
  teardownIntegrationServer
} from '../../utils'

import listPullRequestReviews from '../../fixtures/listPullRequestReviews.json'

describe('list pull request reviews', () => {
  beforeAll(async () => await setupIntegrationServer(listPullRequestReviews))
  afterAll(async () => await teardownIntegrationServer())

  test('displays help', async () => {
    const out = await runCLI({
      args: 'site pull-request-reviews --help'
    })
    expect(out).toMatchSnapshot()
  })

  test('lists pull request reviews', async () => {
    const out = await runCLI({
      args: 'site pull-request-reviews --site=test'
    })
    expect(out).toMatchSnapshot()
  })

  test('lists pull request reviews as json', async () => {
    const out = await runCLI({
      args: 'site pull-request-reviews --site=test --json'
    })
    expect(out).toMatchSnapshot()
  })
})
