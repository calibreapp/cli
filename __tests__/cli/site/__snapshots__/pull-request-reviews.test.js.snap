// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list pull request reviews displays help 1`] = `
"calibre site pull-request-reviews [options]

List Pull Request Reviews for a selected Site.

Options:

  --help     Show help                                                 [boolean]
  --version  Show version number                                       [boolean]
  --site     A unique slug identifying each Site. You can find it in Site Settin
             gs → General or by using the calibre site list command.
                                                             [string] [required]
  --json     Outputs the results of the command in JSON format."
`;

exports[`list pull request reviews lists pull request reviews 1`] = `
"TITLE                    | BRANCH             | SHA      | STATUS    | CREATED           
First Test Pull Request  | author/my-branch   | 241ccbb… | completed | 6:48AM 22-Aug-2023
Second Test Pull Request | author/my-branch-2 | 78f9a44… | completed | 7:31AM 15-Aug-2023"
`;

exports[`list pull request reviews lists pull request reviews as json 1`] = `
"[
  {
    \\"title\\": \\"First Test Pull Request\\",
    \\"status\\": \\"completed\\",
    \\"branch\\": \\"author/my-branch\\",
    \\"sha\\": \\"241ccbb0d2c396c9413adb816fa3ca212aff00da\\",
    \\"createdAt\\": \\"2023-08-22T06:48:03Z\\"
  },
  {
    \\"title\\": \\"Second Test Pull Request\\",
    \\"status\\": \\"completed\\",
    \\"branch\\": \\"author/my-branch-2\\",
    \\"sha\\": \\"78f9a442d4f952b612e2d9627b43a61cd5c7e044\\",
    \\"createdAt\\": \\"2023-08-15T07:31:48Z\\"
  }
]"
`;
